RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc

```diff
@@ -0,0 +1,1711 @@
+
+  Vec3D const threadsPerBlock_addpLSasTrackCandidate{1, 1, 384};
+  Vec3D const blocksPerGrid_addpLSasTrackCandidate{1, 1, max_blocks};
+  WorkDiv3D const addpLSasTrackCandidate_workDiv =
+      createWorkDiv(blocksPerGrid_addpLSasTrackCandidate, threadsPerBlock_addpLSasTrackCandidate, elementsPerThread);
+
+  alpaka::exec<Acc3D>(queue_,
+                      addpLSasTrackCandidate_workDiv,
+                      AddpLSasTrackCandidate{},
+                      nLowerModules_,
+                      trackCandidatesDC_->view(),
+                      segmentsDC_->const_view<SegmentsOccupancySoA>(),
+                      segmentsDC_->const_view<SegmentsPixelSoA>(),
+                      tc_pls_triplets);
+
+  // Check if either n_max_pixel_track_candidates or n_max_nonpixel_track_candidates was reached
+  auto nTrackCanpT5Host_buf = cms::alpakatools::make_host_buffer<unsigned int>(queue_);
+  auto nTrackCanpT3Host_buf = cms::alpakatools::make_host_buffer<unsigned int>(queue_);
+  auto nTrackCanpLSHost_buf = cms::alpakatools::make_host_buffer<unsigned int>(queue_);
+  auto nTrackCanT5Host_buf = cms::alpakatools::make_host_buffer<unsigned int>(queue_);
+  alpaka::memcpy(queue_,
+                 nTrackCanpT5Host_buf,
+                 cms::alpakatools::make_device_view(queue_, (*trackCandidatesDC_)->nTrackCandidatespT5()));
+  alpaka::memcpy(queue_,
+                 nTrackCanpT3Host_buf,
+                 cms::alpakatools::make_device_view(queue_, (*trackCandidatesDC_)->nTrackCandidatespT3()));
+  alpaka::memcpy(queue_,
+                 nTrackCanpLSHost_buf,
+                 cms::alpakatools::make_device_view(queue_, (*trackCandidatesDC_)->nTrackCandidatespLS()));
+  alpaka::memcpy(queue_,
+                 nTrackCanT5Host_buf,
+                 cms::alpakatools::make_device_view(queue_, (*trackCandidatesDC_)->nTrackCandidatesT5()));
+  alpaka::wait(queue_);  // wait to get the values before using them
+
+  auto nTrackCandidatespT5 = *nTrackCanpT5Host_buf.data();
+  auto nTrackCandidatespT3 = *nTrackCanpT3Host_buf.data();
+  auto nTrackCandidatespLS = *nTrackCanpLSHost_buf.data();
+  auto nTrackCandidatesT5 = *nTrackCanT5Host_buf.data();
+  if ((nTrackCandidatespT5 + nTrackCandidatespT3 + nTrackCandidatespLS == n_max_pixel_track_candidates) ||
+      (nTrackCandidatesT5 == n_max_nonpixel_track_candidates)) {
+    printf(
+        "****************************************************************************************************\n"
+        "* Warning: Track candidates were possibly truncated.                                               *\n"
+        "* You may need to increase either n_max_pixel_track_candidates or n_max_nonpixel_track_candidates. *\n"
+        "* Run the code with the WARNINGS flag activated for more details.                                  *\n"
+        "****************************************************************************************************\n");
```

Comment:
I guess this applies only to `printf` outside of the kernel code, right?

Reply:
Yes. Also `printf`s that are behind of a configuration flag (that is off by default) are tolerated. The two that I pointed out looked like they being printed out depends only on the data.

On the other hand, it would be preferable to avoid the use of MessageLogger in `.dev.cc` file. One way would be to abstract the printouts themselves to functions that are implemented in a `.cc` file.